### PR TITLE
feat(storage): get all parquet file list for fuse engine

### LINKED ISSUE #7591: feat: get all parquet file list for fuse engine
**Summary**

```
SELECT * FROM FUSE_SNAPSHOT_FILES(‘<database>‘, ’<table>‘);
```

Returns the parquet file lists with the latest snapshot:
```
snapshot_id, timestamp, file_location
```

To make the downstream work with all parquet files under a snapshot.

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

```
mysql> select * from fuse_block('default','t', '7cc7af2b8fd54222b337ae6912f293ed')\G
*************************** 1. row ***************************
          snapshot_id: 7cc7af2b8fd54222b337ae6912f293ed
            timestamp: 2022-09-13 15:50:50.094912
       block_location: 1/12/_b/ca51137b26874d60a837ab28fa3d4b1e_v0.parquet
           block_size: 8
bloom_filter_location: 1/12/_i/ca51137b26874d60a837ab28fa3d4b1e_v1.parquet
    bloom_filter_size: 223
1 row in set (0.04 sec)
Read 1 rows, 208.00 B in 0.012 sec., 82.15 rows/sec., 16.69 KiB/sec.
```
Fixes #7591

### CODE REVIEW

File: src/query/service/src/procedures/systems/fuse_block.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+#[async_trait::async_trait]
+impl OneBlockProcedure for FuseBlockProcedure {
+    fn name(&self) -> &str {
+        "FUSE_BLOCK"
+    }
+
+    fn features(&self) -> ProcedureFeatures {
+        ProcedureFeatures::default().num_arguments(3)
```
Comment (BohuTANG) [inline src/query/service/src/procedures/systems/fuse_block.rs:44]: I think we should let the user get the snapshot, we can do it with the latest snapshot if the argument is 2.
```
fuse_block('db', 'table')
```

Comment (zhyass) [reply src/query/service/src/procedures/systems/fuse_block.rs]:
```sql
fuse_block('db', 'table'[,'snapshot_id'])
```

**Review by dantengsky — APPROVED** [review]
LGTM

File: src/query/service/tests/it/interpreters/interpreter_call.rs [hunk 2]
```diff
@@ -110,6 +110,72 @@ async fn test_call_fuse_snapshot_interpreter() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn test_call_fuse_block_interpreter() -> Result<()> {
```
[the remark this answers is not in the document]
Comment (BohuTANG) [reply src/query/service/tests/it/interpreters/interpreter_call.rs:114]: The unit test of this PR can't guarantee the logic is right, and someone may break it in the refactoring next time.

[the remark this answers is not in the document]
Comment (zhyass) [reply src/query/service/tests/it/interpreters/interpreter_call.rs:114]: Add

**Pull request merged**